02-Dev-Cluster-Setup/README.md

```diff
@@ -1,178 +1,432 @@
-- `chmod 700 ~/.ssh/id_rsa*`
+### bash
+# change your resource group name and location if desired
 
-If running locally, you can check for existing ssh keys. More info [here](https://docs.github.com/en/github-ae@latest/github/authenticating-to-github/checking-for-existing-ssh-keys).
+export AKDC_LOC=westus2
+export AKDC_RG=akdc
 
-```bash
+# Create a resource group
 
-# for linux
-ls -al ~/.ssh
+az group create -l $AKDC_LOC -n $AKDC_RG
 
-# for windows
-dir %HOMEDRIVE%%HOMEPATH%\.ssh
+### Windows
+
+# change your resource group name and location if desired
+
+set AKDC_LOC=westus2
+set AKDC_RG=akdc
+
+# Create a resource group
+
+az group create -l %AKDC_LOC% -n %AKDC_RG%
 
 ```
 
-If you dont have an existing public or private key pair, generate a new one. More info [here](https://docs.github.com/en/github-ae@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
+## Create Azure VM
+
+> The Azure command takes about 5 minutes to complete
 
 ```bash
 
-ssh-keygen -b 4096 -C "your_email@example.com"
+### bash
+
+# These commands should be executed from 02-Dev-Cluster-Setup/scripts directory.
+
+# replace user name in the setup script
+sed s/ME=akdc/ME=$USER/ startup.sh > akdc.sh
+
+# create an Ubuntu VM and install k8s prerequisites
+# save IP address into the AKDC_IP env var
+
+export AKDC_IP=$(az vm create \
+  -g $AKDC_RG \
+  --admin-username $USER \
+  -n akdc \
+  --size standard_d2s_v3 \
+  --nsg-rule SSH \
+  --image Canonical:UbuntuServer:18.04-LTS:latest \
+  --os-disk-size-gb 128 \
+  --generate-ssh-keys \
+  --query publicIpAddress -o tsv \
+  --custom-data akdc.sh)
+# Notice the path of the akdc.sh file.
+
+rm akdc.sh
+
+# This will output an IP address into the AKDC_IP env var
+echo $AKDC_IP
+
+# (optional) open NodePort range on NSG
+az network nsg rule create -g $AKDC_RG \
+--nsg-name akdcNSG --access allow \
+--description "AKDC Ports" \
+--destination-port-ranges 30000-32767 \
+--protocol tcp \
+-n AkdcPorts --priority 1200
+
+# SSH into the VM
+ssh ${AKDC_IP}
+
+### Windows
+
+# replace user name
+sed s/ME=akdc/ME=%USERNAME%/ startup.sh > akdc.sh
+
+# create an Ubuntu VM and install k8s
+# save IP address into the AKDC_IP env var
+
+for /f %f in (' ^
+  az vm create ^
+  -g %AKDC_RG% ^
+  --admin-username %USERNAME% ^
+  -n akdc ^
+  --size standard_d2s_v3 ^
+  --nsg-rule SSH ^
+  --image Canonical:UbuntuServer:18.04-LTS:latest ^
+  --os-disk-size-gb 128 ^
+  --generate-ssh-keys ^
+  --query publicIpAddress -o tsv ^
+  --custom-data akdc.sh') ^
+do set AKDC_IP=%f
+
+del akdc.sh
+
+echo %AKDC_IP%
+
+# (optional) open NodePort range on NSG
+az network nsg rule create -g %AKDC_RG% ^
+--nsg-name akdcNSG --access allow ^
+--description "AKDC Ports" ^
+--destination-port-ranges 30000-32767 ^
+--protocol tcp ^
+-n AkdcPorts --priority 1200
+
+ssh %AKDC_IP%
```

Comment:
@bartr @kforeverisback
I ran into another issue here in windows cmd. This is failing for me because it appends  `northamerica\\` to my username. I can get around this by doing `ssh %USERNAME%@%AKDC_IP%`. Is there something to do before getting this to work in windows?

Reply:
I don't know of anything you can do to get around that. My username is just "bartr".

I wonder if we should go back to using akdc for simplicity and consistency?  We could also have them set %USER% on Windows (if it's not set).

Reply:
I like using akdc for the name. I tried setting %USER% but was getting the same error.